### ISSUE TITLE
Sage 50 Accounts UK - Creating Invoices with Currency Issue

### ISSUE THREAD

**Issue by nddsoft** [Sage/SData-2.0#160]
Hi,
I'm using Sage Accounts 50 UK version: 24.0.68.0 with SData enabled.

The company is set with base currency GBP, and most customers are also set with currency GBP.
When creating an invoice via SData to a customer in the same GBP currency, the invoice gets generated in GBP as expected.

However, when trying to create an invoice for a customer who was created with a currency of EUR within Sage, the invoice is generated but doesn't convert the GBP to EUR.

The invoice amounts being passed to Sage via SData are in GBP.  When I create an invoice for a EUR currency customer, I tried to  use the line below in the hope to tell SData that the currency used for this invoice is in GBP:
`<crm:currency>GBP</crm:currency>`

However, it does not, and simply assumes the currency used in the invoice is EUR (same as the customer currency).  Thus for example, I'm trying to say that the price of an item is 10 GBP, and Sage shows it as 10 EUR.

This is shown in the response after creating the invoice:
```
<crm:currency>EUR</crm:currency>     // <= As per above, I did set this to GBP, but was ignored?
<crm:operatingCompanyCurrency>GBP</crm:operatingCompanyCurrency>
<crm:operatingCompanyCurrencyExchangeRate>1.11</crm:operatingCompanyCurrencyExchangeRate>
<crm:operatingCompanyCurrencyExchangeRateOperator>/</crm:operatingCompanyCurrencyExchangeRateOperator>
<crm:operatingCompanyCurrencyExchangeRateDate xsi:nil="true" />
```

Basically, how can I tell SData that the invoice being created is in a specific currency?

Thank you for any help.
Steve

**Comment by asterix187** [comment]
Can you create EUR invoices against GBP accounts in the normal Sage 50 client?

________________________________

**Comment by nddsoft** [comment]
Hi Asterix, thank you for replying.

I don't think I can create an invoice in a different currency then the customer's set currency when it was created in Sage. 50 client..
When I create a new invoice for a GBP customer, the invoice defaults to GBP.
When I create a new invoice for a EUR customer, the invoice defaults to EUR.

If the same holds true for SData, then for each non base currency customer, I need to get the currency exchange rate and then convert the price from GBP to match the invoice currency?
I'm not quite sure where the exchange rate can be retrieved from. in SData... 

Thank you for any help.
Steve

**Comment by asterix187** [comment]
Hi, that’s was my thoughts hence my question. Normally people in Sage 50 create a customer record per currency if they trade multi-currency with customers. I don’t have a Sage 50 end point at hand so I am not sure if the currency end point is available but you should be able to check via a $schema call.

**Comment by nddsoft** [comment]
Hi Asterix,
Thanks for that, it helps.

I did check the schema and the endpoints and found I can get the exchange rate that was used for a specific invoice, order, quote and project via this:
`<crm:operatingCompanyCurrencyExchangeRate>1.11</crm:operatingCompanyCurrencyExchangeRate>`
  But I couldn't find a way to extract it from the Sage 50 Settings, as it seems there's one exchange rate set in Sage 50 client (settings > currencies).

It wouldn't be wise to be looking for the latest invoice in EUR that was created and use that exchange rate in a new invoice, as each invoices can have their exchange rate set individually at creation time in Sage 50 Client.

I'll keep looking and update if I find it.

Thank you,
Steve

**Comment by darroncockram** [comment]
Exchange rates are not exposed to the Sdata feeds I'm afraid.

**Comment by nddsoft** [comment]
Hi Darron,
Thank you for the confirmation.
Steve

**Comment by darroncockram** [comment]
That will likely be because  fiddler is not capturing traffic to localhost - https://docs.telerik.com/fiddler/Observe-Traffic/Troubleshooting/NoTrafficToLocalhost
